packages/wallet/contract/src/smartWallet.js

```diff
@@ -4,87 +4,264 @@ import {
-      updateState: value =>
-        publisher.publish({
-          ...assign(mutableState, { [key]: value }),
-        }),
+  // NB: state size must not grow monotonically
+  const myWalletStorageNode = E(storageNode).makeChildNode(address);
+
+  /** @type {PublishKit<PursesRecord>} */
+  const pursePublishKit = makePublishKit();
+  const pursesSubscriber = makeStoredSubscriber(
+    pursePublishKit.subscriber,
+    E(myWalletStorageNode).makeChildNode('purses'),
+    marshaller,
+  );
+
+  const publishPurses = () => {
+    const record = Array.from(brandPurses.entries()).map(([brand, purse]) => {
+      const desc = brandDescriptors.get(brand);
+      return { ...desc, purseBalance: purse.getCurrentAmount() };
     });
+    pursePublishKit.publisher.publish(record);
+  };
+  // #endregion
+
+  /** @type { (desc: BrandDescriptor) => Promise<void>} */
+  const addIssuer = async desc => {
+    // assert haven't received this issuer before.
+    const descriptorsHas = brandDescriptors.has(desc.brand);
+    const pursesHas = brandPurses.has(desc.brand);
+    assert(
+      !(descriptorsHas && pursesHas),
+      'repeated brand from bank asset subscription',
+    );
+    assert(
+      !(descriptorsHas || pursesHas),
+      'corrupted state; one store has brand already',
+    );
+
+    // make a purse for this issuer.
+    // first make the purse so that if anything fails we don't update stores
+    const purse = await E(desc.issuer).makeEmptyPurse();
+
+    // save all five of these in a collection (indexed by brand?) so that when
+    // it's time to take an offer description you know where to get the
+    // relevant purse. when it's time to make an offer, you know how to make
+    // payments. REMEMBER when doing that, need to handle every exception to
+    // put the money back in the purse if anything fails.
+    brandDescriptors.init(desc.brand, desc);
+    brandPurses.init(desc.brand, purse);
+
+    publishPurses();
+  };
+
+  // Ensure a purse for each issuer
+  addIssuer({
+    brand: invitationIssuer.getBrand(),
+    issuer: invitationIssuer,
+    petname: 'invitations',
+  });
+  // watch the bank for new issuers to make purses out of
+  observeIteration(E(bank).getAssetSubscription(), {
+    async updateState(desc) {
+      await addIssuer({
+        brand: desc.brand,
+        issuer: desc.issuer,
+        petname: desc.proposedName,
+      });
+    },
+  });
+  /** @type {ERef<DepositFacet>} */
+  const depositFacet = Far('smart wallet deposit facet', {
+    /**
+     * Put the assets from the payment into the appropriate purse
+     *
+     * @param {Payment} payment
+     * @throws if the purse doesn't exist
+     * ??? PRODUCT: should it instead wait for an issuer/purse to put it in?
+     */
+    receive: payment => {
+      const brand = payment.getAllegedBrand();
+      const purse = brandPurses.get(brand);
+      const depositP = purse.deposit(payment);
+      E.when(depositP, publishPurses);
```

Comment:
use `void` to indicate that we're dropping this promise, right?

Reply:
good catch. will handle in some fashion